### PR TITLE
Add com.francescogaglione.cosmicmoney

### PR DESCRIPTION
**Pull request by francesco-gaglione**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly. 
**Cosmic Money** is a simple and intuitive personal finance manager for the **Cosmic** desktop environment.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: com.francescogaglione.cosmicmoney.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+  "id": "com.francescogaglione.cosmicmoney",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "24.08",
+  "sdk": "org.freedesktop.Sdk",
+  "sdk-extensions": ["org.freedesktop.Sdk.Extension.rust-stable"],
+  "command": "cosmic_money",
+  "finish-args": [
```
Comment (bbhtt) [inline com.francescogaglione.cosmicmoney.json:8]: use the cosmic baseapp like other cosmic apps are doing.

File: com.francescogaglione.cosmicmoney.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    "--share=ipc",
+    "--share=network",
+    "--socket=fallback-x11",
+    "--socket=wayland",
+    "--device=dri",
+    "--talk-name=com.system76.CosmicSettingsDaemon",
+    "--filesystem=xdg-config/cosmic:rw",
+    "--filesystem=xdg-data:rw"
```
Comment (bbhtt) [inline com.francescogaglione.cosmicmoney.json:16]:
```suggestion
```

File: com.francescogaglione.cosmicmoney.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  "sdk-extensions": ["org.freedesktop.Sdk.Extension.rust-stable"],
+  "command": "cosmic_money",
+  "finish-args": [
+    "--share=ipc",
+    "--share=network",
+    "--socket=fallback-x11",
+    "--socket=wayland",
+    "--device=dri",
+    "--talk-name=com.system76.CosmicSettingsDaemon",
+    "--filesystem=xdg-config/cosmic:rw",
```
Comment (bbhtt) [inline com.francescogaglione.cosmicmoney.json:15]: why is rw needed?

Comment (bbhtt) [reply com.francescogaglione.cosmicmoney.json:15]: THis needs an exception, did you run the linter like the docs say?

Comment (bbhtt) [reply com.francescogaglione.cosmicmoney.json:15]: https://docs.flathub.org/docs/for-app-authors/linter#exceptions

File: com.francescogaglione.cosmicmoney.json (hunk 1 above)
Comment (bbhtt) [reply com.francescogaglione.cosmicmoney.json:8]: https://github.com/flathub/dev.edfloreshz.CosmicTweaks/blob/130c94510697a0df4fd49ced4163a4609afd5987/dev.edfloreshz.CosmicTweaks.json#L5-L6

File: com.francescogaglione.cosmicmoney.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
+{
+  "id": "com.francescogaglione.cosmicmoney",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "24.08",
+  "sdk": "org.freedesktop.Sdk",
+  "sdk-extensions": ["org.freedesktop.Sdk.Extension.rust-stable"],
```
Comment (hfiguiere) [inline com.francescogaglione.cosmicmoney.json:6]:
```suggestion
  "sdk-extensions": [
      "org.freedesktop.Sdk.Extension.rust-stable"
  ],
```

File: com.francescogaglione.cosmicmoney.json (hunk 3 above)
Comment (francesco-gaglione) [reply com.francescogaglione.cosmicmoney.json]: sorry it was an oversight

**Comment by francesco-gaglione** [comment]
@bbhtt  solved all

**Comment by bbhtt** [comment]
You still need the send the exceptions PR, see the link I posted and the run the liner on your manifest like the submission docs say. That will give you the error code

**Comment by francesco-gaglione** [comment]
Ah sorry, here it is: https://github.com/flathub-infra/flatpak-builder-lint/pull/527

**Comment by bbhtt** [comment]
There is an infra issue going on so merging the linter PR won't deploy it. I'll merge it once infra is fixed.

**Comment by francesco-gaglione** [comment]
@bbhtt news on this?

**Comment by bbhtt** [comment]
https://status.flathub.org/issues/2024-11-03-repo-exp/

**Comment by bbhtt** [comment]
bot, build com.francescogaglione.cosmicmoney

**Comment by bbhtt** [comment]
```
warning: `/run/build/cosmic_money/cargo/config` is deprecated in favor of `config.toml`
note: if you need to support cargo 1.38 or earlier, you can symlink `config` to `config.toml`
error: failed to get `libcosmic` as a dependency of package `cosmic_money v0.1.0 (/run/build/cosmic_money)`
Caused by:
  failed to load source for dependency `libcosmic`
Caused by:
  Unable to update https://github.com/pop-os/libcosmic.git
Caused by:
  the source https://github.com/pop-os/libcosmic requires a lock file to be present first before it can be
  used against vendored source code
  remove the source replacement configuration, generate a lock file, and then
  restore the source replacement configuration to continue the build
```

**Comment by francesco-gaglione** [comment]
I cannot replicate the problem locally. I am running these commands. Do you think these are the right commands?
```
    flatpak-builder --force-clean --user --install-deps-from=flathub --repo=repo --install builddir com.francescogaglione.cosmicmoney.json --verbose
    flatpak build-bundle repo com.francescogaglione.cosmicmoney.flatpak com.francescogaglione.cosmicmoney --runtime-repo=https://github.com/francesco-gaglione/cosmic-money
 
```

**Comment by bbhtt** [comment]
Commands seem right, but I think something is wrong with the cargo-sources you generated. Is the one here and the one you are locally testing, the same?

**Comment by bbhtt** [comment]
Why are you missing cargo.lock upstream? What are you generating the sources against?

**Comment by francesco-gaglione** [comment]
@bbhtt yes it is the same, copy pasted the file, I'm checking but no issues localy. What do u mean with missing cargo.lock upstream?

**Comment by francesco-gaglione** [comment]
@bbhtt pushed the cargo.lock and generated new sources, see if build

**Comment by bbhtt** [comment]
you can start the build 

bot, build com.francescogaglione.cosmicmoney

**Comment by bbhtt** [comment]
Looks like it worked but you have a misplaced icon


```json
{
    "errors": [
        "non-svg-icon-in-scalable-folder"
    ],
    "info": [
        "non-svg-icon-in-scalable-folder: ['/tmp/tmprcogo5pt/icons/hicolor/scalable/apps/com.francescogaglione.cosmicmoney.png']"
    ],
    "message": "Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter"
}
```

File: com.francescogaglione.cosmicmoney.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        "env": {
+          "CARGO_HOME": "/run/build/cosmic_money/cargo"
+        }
+      },
+      "build-commands": [
+        "cargo --offline build --release --verbose",
+        "install -Dm0755 ./target/release/cosmic_money /app/bin/cosmic_money",
+        "install -Dm0644 ./res/com.francescogaglione.cosmicmoney.desktop /app/share/applications/com.francescogaglione.cosmicmoney.desktop",
+        "install -Dm0644 ./res/com.francescogaglione.cosmicmoney.metainfo.xml /app/share/metainfo/com.francescogaglione.cosmicmoney.metainfo.xml",
+        "install -Dm0644 ./res/icons/hicolor/scalable/apps/com.francescogaglione.cosmicmoney.png /app/share/icons/hicolor/scalable/apps/com.francescogaglione.cosmicmoney.png",
```
Comment (bbhtt) [inline com.francescogaglione.cosmicmoney.json:41]: if you don't have an actual SVG icon then drop this

Comment (hfiguiere) [reply com.francescogaglione.cosmicmoney.json:41]: and fix it upstream too.

Comment (francesco-gaglione) [reply com.francescogaglione.cosmicmoney.json:41]: @bbhtt I don't have an svg, so yes I'll drop it. 
@hfiguiere what do you mean fix it upstream too?

Comment (hfiguiere) [reply com.francescogaglione.cosmicmoney.json:41]: `git rm ./res/icons/hicolor/scalable/apps/com.francescogaglione.cosmicmoney.png`

Comment (francesco-gaglione) [reply com.francescogaglione.cosmicmoney.json:41]: @hfiguiere u mean in my repo (cosmic money) or in this repo?

**Comment by francesco-gaglione** [comment]
bot, build com.francescogaglione.cosmicmoney

**Comment by francesco-gaglione** [comment]
@bbhtt build successful

**Comment by bbhtt** [comment]
I know, I was asleep.

**Comment by francesco-gaglione** [comment]
just to know, for future times... at this point can I cast the merge myself? or is there some rule and does it have to be approved?

**Comment by bbhtt** [comment]
Yea once the repo is created you can merge by yourselves. 

/merge

**Pull request closed without merging**